sla_per_user_system_test.py

```diff
@@ -0,0 +1,501 @@
+        - user190 with 190 shares
+        - user950 qith 950 shares
+        - Each user runs load from own loader (round robin):
+           - user190 runs load with throttle
+           - user950 runs load with high throughput
+        """
+        stress_duration = 10  # minutes
+        shares = [190, 950]
+        max_key_for_read = self.n*0.3
+        read_users = []
+
+        session = self.prepare_schema()
+        self.create_test_data()
+
+        self.fill_cache_before_test(max_key_for_read=max_key_for_read, stress_duration=60)
+
+        # Define Service Levels/Roles/Users
+        for share in shares:
+            read_users.append({'user': User(session=session, name='user%d' % share, password='user%d' % share),
+                               'role': Role(session=session, name='role%d' % share),
+                               'service_level': ServiceLevel(session=session, name='sla%d' % share,
+                                                             service_shares=share)})
+
+        # Create Service Levels/Roles/Users
+        self.create_auths(entities_list_of_dict=read_users)
+
+        read_cmds = [self.define_read_cassandra_stress_command(user=read_users[0], load_type=self.CACHE_ONLY_LOAD,
+                                                               workload_type=self.WORKLOAD_THROUGHPUT, threads=200,
+                                                               stress_duration_min=stress_duration,
+                                                               max_rows_for_cache_read=max_key_for_read),
+                     self.define_read_cassandra_stress_command(user=read_users[1], load_type=self.CACHE_ONLY_LOAD,
+                                                               workload_type=self.WORKLOAD_LATENCY, threads=250,
+                                                               stress_duration_min=stress_duration,
+                                                               max_rows_for_cache_read=max_key_for_read)
+                     ]
+
+        self._throughput_latency_tests_run(read_users=read_users, read_cmds=read_cmds)
+
+    def test_throughput_latency_disk_only_load(self):
+        """
+        Test when one user run load with high latency and another  - with high througput
+        The load is run on the data set that fully exists in the cache
+        Troughput - latency test:
+        - Add SLA and grant to user (before any load)
+        - user190 with 190 shares
+        - user950 qith 950 shares
+        - Each user runs load from own loader (round robin):
+           - user190 runs load with throttle
+           - user950 runs load with high throughput
+        """
+        stress_duration = 10  # minutes
+
+        session = self.prepare_schema()
+        self.create_test_data(rows_amount=300000000)
+
+        max_key_for_cache = self.n*0.3
+        self.fill_cache_before_test(max_key_for_read=max_key_for_cache, stress_duration=30)
```

Comment:
why do we need it?

Reply:
We need it in case cache only test

Reply:
First, this is a test called "disk_only" so I don't understand why we need to "fill the cache" (also bad name, better call it warm the cache).
If the cache is cold every new read will be from disk... so why do we need to warm it in this case?
If we start a sequential read while the cache is cold or warm when relevant data isn't in cache it'll be also "disk_only"

Reply:
I don't see this also

Reply:
I think the cache couldn't be cold because of we run prefill before. Just if we restart the scylla on all nodes

Reply:
Changed function name "fill_cache_before_test" -> "warm_up_cache_before_test"